src/enrollments/enrollments.controller.ts

```diff
@@ -0,0 +1,21 @@
+import { Controller, Post, Body, UseGuards } from "@nestjs/common";
+
+import { Roles } from "@/auth/decorators/roles.decorator";
+import { JwtAuthGuard } from "@/auth/guards/jwt-auth.guard";
+import { RolesGuard } from "@/auth/guards/roles.guard";
+import { EUserType } from "@/common/enums/users.enums";
+
+import { CreateEnrollmentDto, EnrollmentResponseDto } from "./enrollments.dtos";
+import { EnrollmentsService } from "./enrollments.service";
+
+@Controller("enrollments")
+@UseGuards(JwtAuthGuard, RolesGuard)
+export class EnrollmentsController {
+  constructor(private readonly enrollmentsService: EnrollmentsService) {}
+
+  @Roles(EUserType.TEACHER)
+  @Post()
+  enrollStudent(@Body() createEnrollmentDto: CreateEnrollmentDto): Promise<EnrollmentResponseDto> {
+    return this.enrollmentsService.enrollStudent(createEnrollmentDto);
+  }
```

Comment:
I think there should be no enrollments controller at all as it is the junction/pivot table.
Rather this should be an endpoint in the classrooms controller, for example: `classrooms/enroll`

Reply:
Bhaiya I did this for separating enrollments logic from classsroom. There already loads going on in classroom controller, hence I did this for ease.